### DISCUSSION TITLE
Sign a xml create a lot of  X509Certificate

### DISCUSSION THREAD

**Discussion by janux75** [lsh123/xmlsec#704] [General]
Hi
When I signed a xml with a https://www.aleksey.com/xmlsec/api/xmlsec-examples-sign-x509.html#xmlsec-example-sign3 (but using a pkcs12 cert) the xml created four X509Certificate inside a X509Data.

The only modify thing i made is (key_file) is a certificate.pfx:
dsigCtx->signKey = xmlSecCryptoAppPkcs12Load (key_file, "1234", NULL, NULL);
tks

**Comment by lsh123** [comment]
Check the content of your plcs12 file. Likely these 4 certs are there.
